src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -176,57 +125,45 @@ private[spark] object Config {
   val ARMADA_SPARK_EXECUTOR_LABELS: ConfigEntry[String] =
     ConfigBuilder("spark.armada.executor.labels")
       .doc(
-        "A comma separated list of kubernetes labels (in key=value format) to be added to all " +
+        "A comma-separated list of kubernetes labels (in key=value format) to be added to all " +
           "executor pods."
       )
       .stringConf
       .checkValue(k8sLabelListValidator, invalidLabelListErrorMessage)
       .createWithDefaultString("")
 
-  private def k8sLabelListValidator(labelList: String): Boolean = {
-    parseCommaSeparatedK8sLabels(labelList).forall(_.isSuccess)
+  /** Converts a comma-separated list of key=value pairs into a Map.
+    *
+    * @param str
+    *   The string to convert.
+    * @return
+    *   A Map of the key=value pairs.
+    */
+  def commaSeparatedLabelsToMap(str: String): Map[String, String] = {
+    parseCommaSeparatedK8sLabels(str).map(_.get).toMap
   }
-}
 
-object K8sLabelValidator {
-  private val MaxLabelLength  = 63
-  private val MaxPrefixLength = 253
-
-  // DNS-1123 label: lowercase alphanumeric start/end, interior may have '-'
-  private val DNS1123_LABEL = "[a-z0-9]([a-z0-9\\-_/.]*[a-z0-9])?"
-  // DNS-1123 subdomain: series of DNS1123_LABEL separated by '.', total ≤ 253 chars
-  private val DNS1123_SUBDOMAIN = s"$DNS1123_LABEL(?:\\.$DNS1123_LABEL)*"
-
-  // full key = optional prefix "/" name
-  private val LABEL_KEY_REGEX = s"^(?:$DNS1123_SUBDOMAIN/)?$DNS1123_LABEL$$"
-  // value = empty or a DNS1123_LABEL
-  private val LABEL_VALUE_REGEX = s"^$DNS1123_LABEL$$"
-
-  private val keyPattern   = Pattern.compile(LABEL_KEY_REGEX)
-  private val valuePattern = Pattern.compile(LABEL_VALUE_REGEX)
-
-  /** true if `key` is a valid Kubernetes label key (optional DNS1123 subdomain prefix) */
-  def isValidKey(key: String): Boolean = {
-    if (key == null) return false
-
-    // enforce per-segment length limits
-    if (key.contains("/")) {
-      val Array(prefix, name) = key.split("/", 2)
-      if (prefix.length > MaxPrefixLength || name.length > MaxLabelLength) return false
-    } else if (key.length > MaxLabelLength) {
-      // un-prefixed keys are just a label
-      return false
-    }
-
-    keyPattern.matcher(key).matches()
+  private def k8sLabelListValidator(labelList: String): Boolean = {
+    parseCommaSeparatedK8sLabels(labelList).forall(_.isSuccess)
   }
 
-  /** true if `value` is a valid Kubernetes label value (or empty) */
-  def isValidValue(value: String): Boolean = {
-    if (value == null) return false
-    if (value.isEmpty) return true
-    if (value.length > MaxLabelLength) return false
-
-    valuePattern.matcher(value).matches()
+  private def parseCommaSeparatedK8sLabels(str: String): Seq[Try[(String, String)]] = {
+    str.trim
+      .split(",")
+      .map(_.trim)
+      .filter(_.nonEmpty)
+      .map { entry =>
+        entry.trim.split("=", 2).map(_.trim) match {
```

Comment:
I think the two `trim`s in line 153 and 156 (`entry.trim`) are redundant. They call `trim` on the same string.